flink-core/src/main/java/org/apache/flink/api/common/ArchivedExecutionConfig.java

```diff
@@ -54,6 +54,19 @@ public ArchivedExecutionConfig(ExecutionConfig ec) {
 		}
 	}
 
+	public ArchivedExecutionConfig(
```

Comment:
I wasn't mixing things up. For example, in ```Execution#archive()``` we create an ```ArchivedExecution``` through the use of the ```ArchivedExecution(Execution)``` constructor. This pattern is identical for all Archived* classes except the EG (because we need some internal components and dont want to overload the class even more).

Changing the constructor implies changing the ExecutionConfig, which imo is out of scope of this PR. Also, to be consistent, we would also have to touch other ```Archivable``` classes as well.